### DISCUSSION TITLE
Color mode seems to be cached on a page by page basis

### DISCUSSION THREAD

**Discussion by srcarney** [templaza/astroid-framework#921] [General]
Maybe I need to somehow NOT cache the main menu in order to solve this problem?

Basically, I have set up the option to change from light mode to dark mode by clicking on the icon in the top right on my page at https://shalomd1.jdevcloud.com.  If I select the dark mode and then click on some of the pages, then afterwards click to return to light mode, then when I click on all of the recently visited pages, they all open in dark mode even though I selected light mode. That doesn't seem very good to me.  Is dark mode only a page by page setting or should changing from light to dark and visa versa effect all the pages after that point?

Also I am wondering how people have added the switch from light to  dark onto their mobile views.

**Comment by sonvnn** [comment]
It looks like a bug of com_jchoptimize. As it has sorted the appearance order of the ASTROID_COLOR_MODE variable running after the Colormode function is called.
![Screen Shot 2024-12-18 at 10 51 57](https://github.com/user-attachments/assets/d3873cd5-d257-4b99-8740-dbdee47869ed)

**Comment by srcarney** [comment]
Thank you very much Sonny for pointing that out to me. I am happy to hear that is was not caused by your software. I will work on getting a fix to this. The ball is squarely in my court now.

**Comment by srcarney** [comment]
Hello Sonny, I received a surprise answer from JCH Optimize that I need to share with you:

**Begin Comment:**
"This is an issue of page caching. If you visit the page in a particular mode, the page will be cached in that mode so if you visit the page again, you'll see the page in the mode it was cached even if the mode was subsequently changed. This is how page caching works. If you disable JCH Optimize entirely and enable the native page cache plugin, the same happens. 

I have tried ordering the Astroid Plugin after the JCH Optimize plugins and it still doesn't work, so it looks like this function is incompatible with page caching. 

You may need to disable page caching to use this feature. Maybe you can check back with the Astroid developers and see if they can get it to work with page caching. If they can get it to work with the native Page Cache Plugin, it should work with the JCH Optimize Page Cache plugin. If there's something we can do to make our page cache more compatible, I would be happy to explore that." </reply>
**Comment Ended.**

I have to say that the dark mode function is cool, but not cool enough to turn off page caching. 

I am eager to hear your thoughts on this.

Sean

**Reply by srcarney** [reply]
Do you think that JCH Optimize is correct that this plug requires disabling the cache to work?

**Reply by sonvnn** [reply]
You can confirm their comment by disable JCH cache and enable Joomla Cache. If colormode does not work, please let me know. May be I will need you admin account to debug.

**Reply by srcarney** [reply]
I have disabled all JCH extensions and enabled the joomla cache plugin. I the cleared the cache and re-enabled Color Mode in the Astroid Plugin and turn on Color mode in all six of my templates.

The problem does still exist. They way I reproduce it is to turn it to light mode and then click on on some pages. Then I turn on dark mode and click on the same pages. The one I keep testing are the ones under 'scripture' heading/category. When you then click on the links again after changing the mode, it will sometimes display in the wrong mode. I am trying to figure out how to securely give you the login and password.

**Comment by srcarney** [comment]
How can I give you the login and password for the super user account that I just set up for you?

**Reply by sonvnn** [reply]
Please send me via support ticket. Thanks
